src/subdomains/supporting/log/log-job.service.ts

```diff
@@ -333,4 +355,12 @@ export class LogJobService {
     );
     return senderPair ? senderTx.filter((s) => s.id >= senderPair.id) : senderTx;
   }
+
+  private async getCustomBalances(client: EvmClient, assets: Asset[]): Promise<EvmTokenBalance[]> {
+    const customBalances: EvmTokenBalance[] = [];
+    for (const customAddress of Config.financialLog.customAddresses) {
+      customBalances.push(...(await client.getTokenBalances(assets, customAddress)));
+    }
+    return customBalances;
+  }
```

Comment:
You could use `asyncMap`:
```ts
return Util.asyncMap(Config.financialLog.customAddresses, (a) => client.getTokenBalances(assets, a));
```
(the flattening is done in the calling method) or `Promise.all` if you want to run it in parallel.